src/components/Stats.tsx

```diff
@@ -0,0 +1,199 @@
+import { LibraryData } from "../interfaces";
+import { ResponsiveContainer, PieChart, Pie, Sector, BarChart, Bar, CartesianGrid, XAxis, YAxis, Tooltip, Legend } from "recharts";
+
+export interface StatsProps {
+  libraries?: LibraryData[];
+}
+
+export interface StatsState {
+  expanded: boolean;
+  copied: boolean;
+  styled: boolean;
+}
+
+export default class Stats extends React.Component<StatsProps, StatsState> {
+  private statsRef = React.createRef<any>();
+
+  constructor(props: StatsProps) {
+    super(props);
+    this.sortLibraries = this.sortLibraries.bind(this);
+    this.renderListView = this.renderListView.bind(this);
+    this.renderChartView = this.renderChartView.bind(this);
+    this.copy = this.copy.bind(this);
+    this.toggleExpanded = this.toggleExpanded.bind(this);
+    this.toggleFormatting = this.toggleFormatting.bind(this);
+    this.state = { expanded: false, copied: false, styled: true };
+  };
+
+  render(): JSX.Element {
+    let sorted = this.sortLibraries();
+    let tabItems = {
+      "List": this.renderListView(sorted),
+      "Charts": this.renderChartView(sorted)
+    };
+
+    return (
+      <Panel
+        id="stats"
+        headerText={"Aggregate Data"}
+        content={
+          <div className="stats-panel">
+            <Tabs items={tabItems} uniqueId="stats-tabs"/>
+          </div>
+        }
+      />
+    );
+  }
+
+  sortLibraries() {
+    let sorted = {"production": [], "testing": [], "cancelled": []};
+    this.props.libraries && this.props.libraries.forEach((l) => {
+      if (Object.values(l.stages).some(v => v === "cancelled")) {
+        sorted.cancelled.push(l);
+      }
+      else if (Object.values(l.stages).some(v => v === "testing")) {
+        sorted.testing.push(l);
+      }
+      else {
+        sorted.production.push(l);
+      }
+    });
+    return sorted;
+  }
+
+  renderListView(sorted) {
+    let buttons = [
+      <Button
+        key="copy"
+        callback={this.copy}
+        content={this.state.copied ? "Copy Again" : "Copy"}
+        className={"inline squared inverted left-align"}
+      />,
+      <Button
+        key="format"
+        callback={this.toggleFormatting}
+        content={`${this.state.styled ? "Remove" : "Restore"} Formatting`}
+        className="inline squared inverted"
+      />,
+      <Button
+        key="expand"
+        callback={this.toggleExpanded}
+        content={`${this.state.expanded ? "Hide" : "Show"} Library Names`}
+        className="inline squared inverted"
+      />
+    ];
+    return (
+      <div className="list-view">
+        { buttons }
+        <ul
+          className="stats-list"
+          ref={this.statsRef}
+          contentEditable
+          suppressContentEditableWarning
+        >
+          {this.makeLis(sorted)}
+        </ul>
+      </div>
+    );
+  }
+
+
+  renderChartView(sorted) {
+    const colors = {
+      "production": "#809F69",
```

Comment:
Where are these colors coming from? They should be the existing colors.

Reply:
They are the existing colors; they're the colors used for the panels in the library list.  But 1) those colors live inside ReusableComponents (because that's where Panel lives), and 2) those were generated by using SCSS to modify colors with variable names, so that doesn't work inside the `.tsx` file, so I had to figure out the resulting hex codes instead.